src/Microsoft.ML.Transforms/Text/TextCatalog.cs

```diff
@@ -56,8 +56,9 @@ public static class TextCatalog
         /// <param name="catalog">The text-related transform's catalog.</param>
         /// <param name="outputColumnName">Name of the column resulting from the transformation of <paramref name="inputColumnName"/>.</param>
         /// <param name="inputColumnName">Name of the column to transform. If set to <see langword="null"/>, the value of the <paramref name="outputColumnName"/> will be used as source.</param>
-        /// <param name="useMarkerCharacters">Whether to use marker characters to separate words.</param>
-        public static TokenizingByCharactersEstimator TokenizeCharacters(this TransformsCatalog.TextTransforms catalog,
+        /// <param name="useMarkerCharacters">Whether to prepend a marker character, <see langword="0x02"/>, to the beginning,
+        /// and append another marker character, <see langword="0x03"/>, to the end of the output vector of characters.</param>
+        public static TokenizingByCharactersEstimator ProduceCharacterTokens(this TransformsCatalog.TextTransforms catalog,
```

Comment:
>ProduceCharacterTokens [](start = 54, length = 22)

Maybe we should name this one as `ProduceCharacterKeys` ?
See [[https://github.com/dotnet/machinelearning/issues/2864](https://github.com/dotnet/machinelearning/issues/2864)](https://github.com/dotnet/machinelearning/issues/2864)

Or maybe leave this as it is, and call `BreakToWordVector` for word estimator.
It's just a suggestion, would be nice to hear other people, I'm not super great with wordsmithing.

Reply:
I don't feel `ProduceCharacterKeys` is clearer... What is the definition of key under char-level tokenization?

---
In reply to: [264470824](https://github.com/dotnet/machinelearning/pull/2916#discussion_r264470824) [](ancestors = 264470824)

Reply:
It would produce vector of Keys (like you know, MapValueToKey produce, that type of Key).
I agree it's not the best name, but I have some desire to make this two methods slightly distinct since they produce different results.
Or maybe I'm just silly and it's just question of documentation.

---
In reply to: [264479716](https://github.com/dotnet/machinelearning/pull/2916#discussion_r264479716) [](ancestors = 264479716,264470824)

Reply:
I will do `ProduceCharactersAsKeys`.

---
In reply to: [264482253](https://github.com/dotnet/machinelearning/pull/2916#discussion_r264482253) [](ancestors = 264482253,264479716,264470824)